File: 0495-teemo-attacking/0495-teemo-attacking.cpp
class Solution {
public:
    int findPoisonedDuration(vector<int>& timeSeries, int duration) {
        long long n = timeSeries.size();
        long long totalTime = n * duration; 
        long long timeDiff = 0;

        for(int i = 0; i < n - 1; i++) {
            long long timeTaken = timeSeries[i] + duration;
            
            if(timeTaken >= timeSeries[i+1]) {
                timeDiff += timeTaken - timeSeries[i+1];
            }
        }
        return totalTime - timeDiff;
    }
};

File: 0495-teemo-attacking/0495-teemo-attacking.py
class Solution(object):
    def findPoisonedDuration(self, timeSeries, duration):
        """
        :type timeSeries: List[int]
        :type duration: int
        :rtype: int
        """
        n = len(timeSeries)
        if n == 0:
            return 0
            
        totalTime = n * duration
        timeDiff = 0

        for i in range(n - 1):
            timeTaken = timeSeries[i] + duration
            
            if timeTaken >= timeSeries[i + 1]:
                timeDiff += timeTaken - timeSeries[i + 1]
                
        return totalTime - timeDiff
